src/cargo/ops/registry.rs

```diff
@@ -55,6 +59,41 @@ pub fn publish(manifest_path: &Path,
     Ok(())
 }
 
+fn check_git_cleanliness(repo: &git2::Repository) -> CargoResult<()> {
+    let mut opts = git2::StatusOptions::new();
+    opts.include_untracked(true).recurse_untracked_dirs(true);
+    let status = try!(repo.statuses(Some(&mut opts)));
+    let files:Vec<String> = status.iter().map(|entry| {
+        let file = entry.index_to_workdir().unwrap()
+             .old_file().path().unwrap().display();
+        format!("{}", file)
```

Comment:
I moved to some and_then and a match. I think that is a better way to handle it. Let me know if there is a more rust-y way.

Thanks!